### PR TITLE
Add an option to retain JNI IDs after class unloading

### PR DESCRIPTION
**Pull request by gacholio**
In order to measure the impact of keeping jmethodIDs valid for users of AsyncGetCallTrace, add a command line option to retain the JNI ID pools on class unload.

New option is `-XX:[+|-]KeepJNIIDs`

Related: #17466

### CODE REVIEW

**Comment by gacholio** [comment]
jenkins test sanity win jdk8

**Comment by gacholio** [comment]
jenkins test sanity win jdk8

File: runtime/vm/classallocation.c [hunk 1]
```diff
@@ -376,8 +376,11 @@ freeClassLoader(J9ClassLoader *classLoader, J9JavaVM *javaVM, J9VMThread *vmThre
 		RELEASE_CLASS_LOADER_BLOCKS_MUTEX(javaVM);
 	}
 
-	if (classLoader->jniIDs != NULL) {
-		pool_kill(classLoader->jniIDs);
+	if (NULL != classLoader->jniIDs) {
```
Comment (tajila) [inline runtime/vm/classallocation.c:379]: dont we still need to clear them so they dont point to garbage memory?

Comment (gacholio) [reply runtime/vm/classallocation.c:379]: This isn't real - it's solely to measure the overhead of not freeing the IDs.

Comment (tajila) [reply runtime/vm/classallocation.c:379]: ah okay

**Pull request merged**